### ISSUE TITLE
#1475 user auth (step 2 extended) - user privilege groups

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc#1520]
#1480 distinguishes between two user types: "authenticated" (with passphrase) and "public" (no passphrase).  Global/suite config can set the privilege level for public users, and authenticated users get full control privileges.  Public read-only is now an option, but is probably not advisable at a large site because of the potential for hammering suite daemons with GUI info requests.

The new connection validator receives the client user and host names (and UUID) as well as passphrase, so we easily could split both public and authenticated users into multiple groups according to users and/or hosts specified under the new config [authentication] sections.  So for instance, you could allow a specified subset of all users, or of users on a particular host, to have read-only access to a suite, and allow only "identity" information to others.  Or you could give a subset of authenticated (with passphrase) users read-only access.

This would be almost as good as the proposed final "user account" solution (step 3 of #1475) but it could be implemented very quickly (est. 1 hour after #1480) and would give us some freedom to think through the final solution carefully.

**Comment by hjoliver** [comment]
(@cylc/core - let me know if you agree with this; I won't do anything with it till then)

**Comment by matthewrmshin** [comment]
(Thinking about this right now.)

**Comment by hjoliver** [comment]
A use for this that I hadn't thought of - until #164 is done only the suite owner should be able to use the gcylc graph view (although this could be hardwired, without any additional global or suite config).

**Comment by matthewrmshin** [comment]
As an interim solution, this is probably OK, (but it does remind me of the old `rsh` and its `.rhosts` file). Eventually, we should move towards a security token system or a public-private key system for access control.

**Comment by hjoliver** [comment]
Agreed; it's definitely just an interim solution.  It'll quickly give us the ability to assign non-global read-only access to suites.

**Comment by hjoliver** [comment]
This interim solution may not be worth doing. Superseded by #1901.

**Comment by hjoliver** [comment]
(besides, this would not work via a web interface, where the "username" associated with the client connection, e.g. in a browser could easily be entirely unrelated to the Unix user accounts on the suite host).
